### ISSUE TITLE
Content migration - add "Deploying to GitLab Pages" doc

### ISSUE THREAD

**Issue by amberleyromo** [gatsbyjs/gatsby#8738]
## Issue Context

This issue is part of the content migration project (#8103), which is meant to take existing content and repurpose it so it's easier to find in the docs.

### Motivation

We are doing this because many people have written excellent content that we want to canonize and maintain so more people can find it and benefit from it.

## Steps to resolve this issue

1. Read the Style Guide and follow it in order for your PR to be accepted. Pay special attention to the templates. 
- Style Guide: https://www.gatsbyjs.org/docs/gatsby-style-guide/
- Docs Templates: https://www.gatsbyjs.org/docs/templates/
- Example of a guide to imitate: https://www.gatsbyjs.org/docs/deploying-to-s3-cloudfront

2. Create the "Deploying to GitLab Pages" article by creating a file named `deploying-to-gitlab-pages.md` in [this directory](https://github.com/gatsbyjs/gatsby/tree/master/docs/docs).

3. Copy and paste this section of the deploying doc: https://www.gatsbyjs.org/docs/deploy-gatsby/#gitlab-pages

4. Make sure you adjust the content to follow the format listed in these resources:
- Docs Templates: https://www.gatsbyjs.org/docs/templates/
- Example of a guide to imitate: https://www.gatsbyjs.org/docs/deploying-to-s3-cloudfront

5. Link the new doc in the docs sidebar file, [doc-links.yml](https://github.com/gatsbyjs/gatsby/blob/master/www/src/data/sidebars/doc-links.yaml), under the Deploying section:

```yaml
- title: Deploying to GitLab Pages
  link: /docs/deploying-to-gitlab-pages
```

**Comment by amberleyromo** [comment]
Open PR: https://github.com/gatsbyjs/gatsby/pull/8731

**Comment by amberleyromo** [comment]
Closed in #8731
